skore/tests/unit/sklearn/train_test_split/test_train_test_split.py

```diff
@@ -182,30 +182,78 @@ def test_train_test_split_kwargs():
     assert output1 == output2
 
 
+def test_train_test_split_single_posargs():
+    """Passing single positional argument with as_dict=True should return as dict."""
+
+    X = [[1]] * 20
+
+    output = train_test_split(X, as_dict=True)
+    # assert that first key contains X
+    assert next(iter(output)).startswith("X")
+    # assert that output is a dict
+    assert type(output) is dict
+
+
+def test_train_test_split_two_posargs():
+    """Passing two positional argument with as_dict=True should return as dict."""
+
+    X = [[1]] * 20
+    y = [0] * 10 + [1] * 10
+
+    output = train_test_split(X, y, as_dict=True)
+    keys = list(output.keys())
+    # assert that keys have X and y
+    assert keys[0].startswith("X") and keys[2].startswith("y")
+    # assert that output is a dict
+    assert type(output) is dict
+
+
+def test_train_test_split_mix_args():
+    """Passing mixed positional and keyword argument with as_dict=True
+    should return as dict."""
+
+    X = [[1]] * 20
+    y = [0] * 10 + [1] * 10
+    z = [0] * 10 + [1] * 10
+
+    output = train_test_split(X, y, z=z, as_dict=True)
+    keys = list(output.keys())
+    # assert that keys have X, y, z
+    assert (
+        keys[0].startswith("X") and keys[2].startswith("y") and keys[4].startswith("z")
+    )
+    # assert that output is a dict
+    assert type(output) is dict
+
+
 def test_train_test_split_dict_kwargs():
-    """Passing data without keyword arguments with return_dict=True
+    """Passing three or more keyword arguments with as_dict=True
     should raise ValueError."""
 
     X = [[1]] * 20
     y = [0] * 10 + [1] * 10
+    z = [0] * 10 + [1] * 10
+
+    err_msg = (
+        "With as_dict=True, more than two arguments must be keyword arguments."
+        "\nEx: train_test_split(X=X, y=y, z=z, sw=sample_weight, as_dict=True)"
+    )
 
-    with pytest.raises(
-        ValueError,
-        match="When as_dict=True, arrays must be passed as keyword arguments",
-    ):
-        train_test_split(X, y, random_state=0, as_dict=True)
+    with pytest.raises(ValueError) as err:
+        train_test_split(X, y, z, random_state=0, as_dict=True)
+    assert str(err.value) == err_msg
```

Comment:
use `raises`'s `match` argument
```suggestion
    with pytest.raises(ValueError, match=err_msg):
        train_test_split(X, y, z, random_state=0, as_dict=True)
```